examples/generative/text_generator.exs

```diff
@@ -0,0 +1,124 @@
+# Based on https://machinelearningmastery.com/text-generation-lstm-recurrent-neural-networks-python-keras/
+Mix.install([
+  {:axon, github: "elixir-nx/axon"},
+  {:nx, "~> 0.2.1"},
+  {:exla, "~> 0.2.2"},
+  {:req, "~> 0.3.0"}
+])
+
+EXLA.set_as_nx_default([:tpu, :cuda, :rocm, :host])
+
+defmodule TextGenerator do
+  require Axon
+
+  @download_url "https://www.gutenberg.org/files/11/11-0.txt"
+  @filename "alice_in_wonderland.txt"
+  @sequence_length 100
+  @batch_size 128
+
+  def book_text do
+    unless File.exists?(@filename) do
+      Req.get!(@download_url)
+      |> then(fn %{body: content} -> File.write!(@filename, content) end)
+    end
+
+    File.read!(@filename)
+  end
+
+  def build_model(characters_count) do
+    Axon.input({nil, @sequence_length, 1}, "input_chars")
+    |> Axon.lstm(256)
+    |> then(fn {_, out} -> out end)
+    |> Axon.nx(fn x -> Nx.slice_along_axis(x, @sequence_length - 1, 1, axis: -2) end)
+    |> Axon.dropout(rate: 0.2)
+    |> Axon.flatten()
+    |> Axon.dense(characters_count, activation: :softmax)
+  end
+
+  def generate(model, params, init_seq, char_to_int, characters) do
+    init_seq =
+      init_seq
+      |> String.downcase()
+      |> String.to_charlist()
+
+    Enum.reduce(1..80, init_seq, fn _, seq ->
+      init_seq =
+        seq
+        |> Enum.reverse()
+        |> Enum.take(@sequence_length)
+        |> Enum.reverse()
```

Comment:
```suggestion
        |> Enum.take(-@sequence_length)
```

Reply:
Is this because of the character we add at the end of each iteration? If so, then we can shift it there, like `tl(seq) ++ [char]`.

Reply:
The first suggestion is better because in the second one we will loose some parts of the sentence especially if we generate more than sequence length.

Reply:
Addressed in 0e6b697